File: app/src/test/java/com/zalora/wajahat/twitsplit/mvp/presenters/SendPresenterTest.java
package com.zalora.wajahat.twitsplit.mvp.presenters;

import com.zalora.wajahat.twitsplit.R;
import com.zalora.wajahat.twitsplit.mvp.impl.SendImpl;
import com.zalora.wajahat.twitsplit.mvp.interactors.SendInteractor;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class SendPresenterTest {

    @Mock
    private SendImpl impl;
    @Mock
    private SendInteractor interactor;
    private SendPresenter presenter;

    @Before
    public void setUp() throws Exception {
        presenter = new SendPresenter(impl, interactor);
    }

    @Test
    public void showErrorWhenTweetIsEmpty() throws Exception {
        when(impl.getTweet()).thenReturn("");
        presenter.sendTweet();
        verify(impl).showValidationError(R.string.tweet_empty);
    }
}

File: app/src/test/java/com/zalora/wajahat/twitsplit/mvp/presenters/SendPresenterTest.kt
package com.zalora.wajahat.twitsplit.mvp.presenters

import com.zalora.wajahat.twitsplit.R
import com.zalora.wajahat.twitsplit.mvp.impl.SendImpl
import com.zalora.wajahat.twitsplit.mvp.interactors.SendInteractor
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.`when`
import org.mockito.Mockito.verify
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class SendPresenterTest {

    @Mock
    private val impl: SendImpl? = null
    @Mock
    private val interactor: SendInteractor? = null
    private var presenter: SendPresenter? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        presenter = SendPresenter(impl!!, interactor!!)
    }

    @Test
    @Throws(Exception::class)
    fun showErrorWhenTweetIsEmpty() {
        `when`(impl!!.getTweet()).thenReturn("")
        presenter!!.sendTweet()
        verify(impl).showValidationError(R.string.tweet_empty)
    }
}